frontend/js/main.js

```diff
@@ -0,0 +1,2203 @@
+        tableCell.setAttribute('id', 'dataTypeKey' + (i + 1));
+      }
+      else if (j == 1) {
+        tableCell.setAttribute('id', 'dataTypeVal' + (i + 1));
+        optionsLength = globalDataTypeList[Object.keys(globalDataTypeList)[i]].length;
+        dataTypeArr = [];
+        for (var k = 0; k < optionsLength; k++) {
+          dataTypeArr.push(globalDataTypeList[Object.keys(globalDataTypeList)[i]][k].T);
+        }
+
+        selectHTML = '';
+        selectId = 'dataTypeOption' + (i + 1);
+        selectHTML = `<div style='display: flex;'>
+                        <i class="large material-icons warning" style='cursor: pointer; visibility: hidden;'>warning</i>
+                        <select onchange='dataTypeUpdate(id)' class='form-control tableSelect' id=${selectId} style='border: 0px !important;'>
+                      </div>`;
+        for (var k = 0; k < optionsLength; k++) {
+          if (k == 0 && globalDataTypeList[Object.keys(globalDataTypeList)[i]][k].Brief !== "") {
+            selectHTML = `<div style='display: flex;'>
+                            <i class="large material-icons warning" style='cursor: pointer;' data-toggle='tooltip' data-placement='bottom' title='${globalDataTypeList[Object.keys(globalDataTypeList)[i]][k].Brief}'>warning</i>
+                            <select onchange='dataTypeUpdate(id)' class='form-control tableSelect' id=${selectId} style='border: 0px !important; font-family: FontAwesome;'>
+                          </div>`;
+          }
+          selectHTML += `<option value='${globalDataTypeList[Object.keys(globalDataTypeList)[i]][k].T}'>${globalDataTypeList[Object.keys(globalDataTypeList)[i]][k].T} </option>`;
+        }
+        selectHTML += `</select>`;
+        tableCell.innerHTML = selectHTML;
+      }
+      tableRow.appendChild(tableCell);
+    }
+    tableBody.appendChild(tableRow);
+  }
+  globalDataTypeTable.appendChild(tableBody);
+  globalDataTypeDiv = document.getElementById('globalDataType');
+  globalDataTypeDiv.innerHTML = '';
+  globalDataTypeDiv.appendChild(globalDataTypeTable);
+  tooltipHandler();
+}
+
+/**
+ * Function to update data types with warnings(if any) in global data type table
+ *
+ * @param {string} id id of select box in global data type table
+ * @return {null}
+ */
+const dataTypeUpdate = (id) => {
+  idNum = parseInt(id.match(/\d+/), 10);
+  dataTypeOptionArray = globalDataTypeList[document.getElementById('dataTypeKey' + idNum).innerHTML];
+
+  optionHTML = '';
+  selectHTML = `<div style='display: flex;'>
+                  <i class="large material-icons warning" style='cursor: pointer; visibility: hidden;'>warning</i>
+                  <select onchange='dataTypeUpdate(id)' class='form-control tableSelect' id=${id} style='border: 0px !important;'>
+                </div>`;
+  for (var x = 0; x < dataTypeOptionArray.length; x++) {
+    optionFound = false;
+    if (dataTypeOptionArray[x].T == document.getElementById(id).value) {
```

Comment:
The if statement can be simplified as below

`optionFound = dataTypeOptionArray[x].T == document.getElementById(id).value;`

Reply:
Agreed.. if condition is simplified now